apisix/plugins/jwt-auth.lua

```diff
@@ -186,8 +189,14 @@ function _M.rewrite(conf, ctx)
     end
     core.log.info("consumer: ", core.json.delay_encode(consumer))
 
-    local auth_secret = get_secret(consumer.auth_conf)
-    jwt_obj = jwt:verify_jwt_obj(auth_secret, jwt_obj)
+    if not consumer.auth_conf.algorithm or consumer.auth_conf.algorithm == "HS256"
+            or consumer.auth_conf.algorithm == "HS512" then
+        local auth_secret = get_secret(consumer.auth_conf)
+        jwt_obj = jwt:verify_jwt_obj(auth_secret, jwt_obj)
+    end
+    if consumer.auth_conf.algorithm and consumer.auth_conf.algorithm == "RS256" then
```

Comment:
code style: `if consumer.auth_conf.algorithm == "RS256" then` is enough

Reply:
and I think we can add a blank line before this line

Reply:
resolve it